.gitignore

```diff
@@ -1,19 +1,51 @@
+# TODO
+TODO.md
+
+# package lock
```

Comment:
package lock is necessary (i know we don't use the npm and pnpm) see: https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5, but i don't think add the package lock is good idea.

Reply:
You had talked the reason of ignoring package lock files. I agree it. It is very important.

I think we should develop this repo only by yarn.

`package-lock.json` is ignored before this pr. 

![image](https://github.com/user-attachments/assets/13405876-423a-42f4-bc6e-e3d1458e4616)

Reply:
Someone use `npm` pr `pnpm` is ok. The files generated by executing `xxx install` should be ignored. Otherwise, the pr submitted by developers will include `package-lock.json` or `pnpm-lock.yaml`.